controllers/cards.js

```diff
@@ -1,70 +1,62 @@
+/* eslint-disable import/no-dynamic-require */
 /* eslint-disable consistent-return */
 /* eslint-disable no-unused-vars */
 const path = require('path');
 const validator = require('validator');
-// eslint-disable-next-line import/no-dynamic-require
+
 const Card = require(path.resolve('models/card.js'));
+const BadRequestError = require(path.resolve('errors/BadRequestError.js'));
+const NotFoundError = require(path.resolve('errors/NotFoundError.js'));
+const UnauthorizedError = require(path.resolve('errors/UnauthorizedError.js'));
 
-module.exports.getCards = (req, res) => {
+module.exports.getCards = (req, res, next) => {
   Card.find({})
-    .then((card) => res.send({ data: card }))
-    .catch((err) => res.status(500).send({ message: 'Произошла ошибка' }));
+    .then((card) => { res.send({ data: card }); })
+    .catch(next);
 };
 
-module.exports.createCard = (req, res) => {
+module.exports.createCard = (req, res, next) => {
   const { name, link } = req.body;
   if (!validator.isURL(link)) {
-    return res.status(500).send({ message: 'invalid URL' });
+    throw new BadRequestError('invalid URL');
   }
   Card.create({ name, link, owner: req.user._id })
     .then((card) => res.send({ data: card }))
-    .catch((err) => res.status(500).send({ message: 'Произошла ошибка' }));
+    .catch(next);
 };
 
-module.exports.deleteCard = (req, res) => {
-  Card.findByIdAndRemove(req.params.cardId)
+module.exports.deleteCard = (req, res, next) => {
+  Card.findById(req.params.Id)
     .then((card) => {
-      if (!card) { throw new Error('Source not found'); }
+      if (!card) { throw new NotFoundError('Source not found'); }
       return card;
     })
-    .then((card) => res.send({ data: card }))
-    .catch((err) => {
-      if (err.name === 'CastError' || err.message === 'Source not found') {
-        res.status(404).send({ message: 'Ресурс не найден!' });
-      } else {
-        res.status(500).send({ message: 'Произошла ошибка' });
+    .then((card) => {
+      if (card.owner.toString() !== req.user._id) {
+        throw new UnauthorizedError('Unauthorized!');
       }
-    });
+    })
+    .then((card) => Card.deleteOne(card))
+    .then((card) => res.send({ data: card }))
```

Comment:
Можно лучше: при удалении карточки возвращать старое состояние или сообщение.
Метод `.deleteOne()` возвращает информацию об удалении записи:
```json
{
    "data": {
        "n": 1,
        "ok": 1,
        "deletedCount": 1
    }
}
```
Это не очень информативно на клиенте, лучше изменить код так, чтобы использовать предыдущий `card`:
```suggestion
    .then((card) => Card.deleteOne(card).then(() => res.send({ data: card }))
```
Либо вместо данных отправлять сообщение, что карточка удалена.

Reply:
+